src/target/riscv/riscv.c

```diff
@@ -1614,6 +1614,282 @@ static int riscv_hit_trigger_hit_bit(struct target *target, uint32_t *unique_id)
 	return ERROR_OK;
 }
 
+static inline uint32_t get_bits_field(riscv_insn_t b, uint32_t lo, uint32_t len) {
+	return (b >> lo) & (((uint32_t)1 << len) - 1);
+}
+
+static inline int16_t get_imm_clwsp(riscv_insn_t instruction) {
+	return (get_bits_field(instruction, 4, 3) << 2)
+		   + (get_bits_field(instruction, 12, 1) << 5)
+		   + (get_bits_field(instruction, 2, 2) << 6);
+}
+
+static inline int16_t get_imm_cldsp(riscv_insn_t instruction) {
+	return (get_bits_field(instruction, 5, 2) << 3)
+		   + (get_bits_field(instruction, 12, 1) << 5)
+		   + (get_bits_field(instruction, 2, 3) << 6);
+}
+
+static inline int16_t get_imm_cswsp(riscv_insn_t instruction) {
+	return (get_bits_field(instruction, 9, 4) << 2)
+		   + (get_bits_field(instruction, 7, 2) << 6);
+}
+
+static inline int16_t get_imm_csdsp(riscv_insn_t instruction) {
+	return (get_bits_field(instruction, 10, 3) << 3)
+		   + (get_bits_field(instruction, 7, 3) << 6);
+}
+
+static inline int16_t get_imm_clw(riscv_insn_t instruction) {
+	return (get_bits_field(instruction, 6, 1) << 2)
+	       + (get_bits_field(instruction, 10, 3) << 3)
+		   + (get_bits_field(instruction, 5, 1) << 6);
+}
+
+static inline int16_t get_imm_cld(riscv_insn_t instruction) {
+	return (get_bits_field(instruction, 10, 3) << 3)
+		   + (get_bits_field(instruction, 5, 2) << 6);
+}
+
+static inline int16_t get_imm_clq(riscv_insn_t instruction) {
+	return (get_bits_field(instruction, 11, 1) << 4)
+		   + (get_bits_field(instruction, 12, 1) << 5)
+		   + (get_bits_field(instruction, 5, 2) << 6)
+		   + (get_bits_field(instruction, 10, 1) << 8);
+}
+
+static inline int16_t get_imm_csw(riscv_insn_t instruction) {
+	return (get_bits_field(instruction, 6, 1) << 2)
+		   + (get_bits_field(instruction, 10, 3) << 3)
+		   + (get_bits_field(instruction, 5, 1) << 6);
+}
+
+static inline uint32_t get_rs1_c(riscv_insn_t instruction) {
+	return 8 + get_bits_field(instruction, 7, 3);
+}
+
+static inline int get_memaddr_storeload(struct target *target,
+		const riscv_insn_t instruction, target_addr_t *mem_addr_wp) {
+	/* find out which memory address is accessed by the instruction at dpc */
+	/* opcode is first 7 bits of the instruction */
+	uint32_t opcode = instruction & 0x7F;
+	/* RVxxC MASK_C = 0xe003 for load/store instructions */
+	if ((instruction & 0x03) < 0x03) {
+		opcode = instruction & 0xe003;
+	}
+	uint32_t rs1;
+	int16_t imm;
+	target_addr_t mem_addr = 0;
+	bool misa_f = riscv_supports_extension(target, 'F');
+	bool misa_d = riscv_supports_extension(target, 'D');
+
+	switch (opcode) {
+	case MATCH_LB: // and MATCH_SB:
```

Comment:
```suggestion
	case MATCH_LB:
	/* fallthrough */
	case MATCH_SB:
```
Is this the intention?

Reply:
Yes. I fixed it in the commit https://github.com/riscv-collab/riscv-openocd/pull/1029/commits/fd7c9b21a2725f47c5912dbc00711b3a67b9c205